homeassistant/components/drop_connect/entity.py

```diff
@@ -0,0 +1,53 @@
+"""Base entity class for DROP entities."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    CONF_DEVICE_DESC,
+    CONF_DEVICE_NAME,
+    CONF_DEVICE_OWNER_ID,
+    CONF_DEVICE_TYPE,
+    CONF_HUB_ID,
+    DEV_HUB,
+    DOMAIN,
+)
+from .coordinator import DROPDeviceDataUpdateCoordinator
+
+
+class DROPEntity(CoordinatorEntity[DROPDeviceDataUpdateCoordinator]):
```

Comment:
Side note: A possible improvement in the future: There's no availability check for the entity at the moment, since the coordinator doesn't poll and we haven't implemented anything specific.

The minimum would probably be to use the `async_subscribe_connection_status` helper in the MQTT integration to subscribe to connection status changes of this client to the broker. The second step would be that each device would have a way to let clients know that it's gone unavailable. That probably requires some logic in the 3rd party library.

The tasmota integration has examples of both things.

The `available` entity property is documented here:
https://developers.home-assistant.io/docs/core/entity#generic-properties

Reply:
There will definitely be future improvement PRs, and this concept is already on my to-do list.  The problem is that the DROP hub should (ideally) never disconnect, and if it does it is likely to be a network problem or a power loss with a missing backup battery.  In those cases, the DROP hub will not be able to publish an offline transition, so I probably need a polling mechanism on the HA side to determine when the hub is offline.  Regardless, there are other ways in our product ecosystem for the user to learn when the devices are offline, so if the data in HA goes stale it will not be unexpected.